semantic/types.go

```diff
@@ -65,6 +65,7 @@ const (
 	Array
 	Object
 	Function
+	TypeVar
```

Comment:
This seems odd. Why is a type variable a first class type within Flux itself? No value can have a type of TypeVar.

Reply:
I don't have a good answer for this at the moment so this will probably change.